### PR TITLE
Kmeans

### PR DESCRIPTION
**Pull request by BioTong**
Add functions related to k-means clustering, unit test script, and description files.

### CODE REVIEW

File: R/kmeans.R [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+#' a mutational discovery or prediction tool.
+#' @param ... Arguments of \code{\link[stats]{kmeans}} function.
+#' @return A one-column data frame with sample IDs as row names and cluster number for each sample.
+#' @seealso \link[stats]{kmeans}
+#' @examples 
+#' data(res_annot)
+#' k_out <- withr::with_seed(123, Kmeans(res_annot, centers = 2))
+#' @export
+
+Kmeans <- function(result, ...){
```
Comment (achevali) [inline R/kmeans.R:14]: Just keep it lower case ```kmeans```

**Review by achevali — CHANGES_REQUESTED** [review]

File: R/kmeans.R [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                            names_to = "signature",
+                            values_to = "exposure",
+                            names_repair = "minimal")
+      ggplot2::ggplot(clust_by_sigs, aes_string(x = "UMAP_1", y = "UMAP_2", colour = "exposure")) +
+        geom_point() +
+        facet_grid(cluster ~ signature) +
+        ggplot2::scale_colour_gradientn(colors = c("blue","green","yellow","orange","red"), name = "Fraction")
+    }
+    else if(group == "annotation"){
+      if(ncol(result@musica@sample_annotations) == 1) {stop("Sample annotation not found.")}
```
[the remark this answers is not in the document]
Comment (BioTong) [reply R/kmeans.R:73]: I find that the sample_annotations data.frame in the musica object has one and only one column of sample IDs initially, so I guess checking the number of columns can tell you if sample annotation is added?

[the remark this answers is not in the document]
Comment (achevali) [reply R/kmeans.R:73]: You should have annotation as a parameter defaulting to NULL, and if group == "annotation", check that ```annotation %in% names(samp_annot(result))```

**Comment by achevali** [comment]
Push any updates you've made to the code so I can review them.

**Pull request closed without merging**